WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -0,0 +1,333 @@
+			.OnEntry(() =>
+			{
+				PlayVisible = false;
+				StopVisible = true;
+				CurrentStatus = _coinJoiningMessage;
+				coinJoinManager.Start(walletVm.Wallet);
+			})
+			.OnProcess(UpdateWalletMixedProgress);
+
+		_machine.Configure(State.ManualFinished)
+			.SubstateOf(State.ManualCoinJoin)
+			.Permit(Trigger.Play, State.ManualPlaying)
+			.OnEntry(() =>
+			{
+				StopVisible = false;
+				PlayVisible = true;
+				CurrentStatus = _finishedMessage;
+				ProgressValue = 100;
+				ElapsedTime = "";
+				RemainingTime = "";
+			});
+
+		// AutoCj State
+		_machine.Configure(State.AutoCoinJoin)
+			.Permit(Trigger.AutoCoinJoinOff, State.ManualCoinJoin)
+			.Permit(Trigger.AutoCoinJoinEntered, State.AutoStarting)
+			.OnEntry(() =>
+			{
+				IsAuto = true;
+				StopVisible = false;
+				PauseVisible = false;
+				PlayVisible = true;
+
+				CurrentStatus = _initialisingMessage;
+
+				coinJoinManager.Stop(walletVm.Wallet);
+				coinJoinManager.AutoStart(walletVm.Wallet);
+			});
+
+		_machine.Configure(State.AutoStarting)
+			.SubstateOf(State.AutoCoinJoin)
+			.Permit(Trigger.Pause, State.Paused)
+			.Permit(Trigger.RoundStart, State.AutoPlaying)
+			.Permit(Trigger.Play, State.AutoPlaying)
+			.OnEntry(() =>
+			{
+				IsAutoWaiting = true;
+				CurrentStatus = _countDownMessage;
+			})
+			.OnProcess(() =>
+			{
+				ElapsedTime = $"{DateTime.Now - _countDownStarted:mm\\:ss}";
+				RemainingTime = $"-{_autoStartTime - DateTime.Now:mm\\:ss}";
+
+				var total = (_autoStartTime - _countDownStarted).TotalSeconds;
+				var percentage = (DateTime.Now - _countDownStarted).TotalSeconds * 100 / total;
+				ProgressValue = percentage;
+			})
+			.OnExit(() =>
+			{
+				IsAutoWaiting = false;
+			});
+
+		_machine.Configure(State.Paused)
+			.SubstateOf(State.AutoCoinJoin)
+			.Permit(Trigger.Play, State.AutoPlaying)
+			.OnEntry(() =>
+			{
+				IsAutoWaiting = true;
+
+				CurrentStatus = _pauseMessage;
+				ProgressValue = 0;
+
+				PauseVisible = false;
+				PlayVisible = true;
+
+				coinJoinManager.Stop(walletVm.Wallet);
+			})
+			.OnProcess(UpdateWalletMixedProgress);
+
+		_machine.Configure(State.AutoPlaying)
+			.SubstateOf(State.AutoCoinJoin)
+			.Permit(Trigger.Pause, State.Paused)
+			.Permit(Trigger.PlebStop, State.Paused)
+			.Permit(Trigger.RoundStartFailed, State.AutoFinished)
+			.Permit(Trigger.RoundStart, State.AutoPlaying)
+			.OnEntry(() =>
+			{
+				IsAutoWaiting = false;
+				PauseVisible = true;
+				PlayVisible = false;
+				CurrentStatus = _coinJoiningMessage;
+				coinJoinManager.Start(walletVm.Wallet);
+			})
+			.OnProcess(UpdateWalletMixedProgress);
+
+		_machine.Configure(State.AutoFinished)
+			.SubstateOf(State.AutoCoinJoin)
+			.Permit(Trigger.RoundStart, State.AutoPlaying)
+			.OnEntry(() =>
+			{
+				PauseVisible = false;
+				PlayVisible = true;
+
+				ProgressValue = 100;
+				ElapsedTime = "";
+				RemainingTime = "";
+
+				CurrentStatus = _finishedMessage;
+			});
```

Comment:
This instantiation and configuration could be extracted to another method to keep the ctor cleaner

Reply:
done.